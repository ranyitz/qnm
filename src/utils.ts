export function isTruthy<T>(x: T | undefined | null): x is T {
  return x !== undefined && x !== null;
}
