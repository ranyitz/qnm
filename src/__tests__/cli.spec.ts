import { execSync } from 'child_process';
import { resolveFixture } from './utils';

const qnmBin = require.resolve('../../bin/qnm');

const runCommand = (
  command: string,
  { cwd, env }: { cwd: string; env?: Record<string, any> },
) =>
  execSync(`${qnmBin} ${command}`, {
    cwd,
    env: {
      ...process.env,
      ...env,
    },
    encoding: 'utf-8',
  });

describe('CLI', () => {
  describe('qnm with no arguments', () => {
    it('should show the version on a single module when called with a string', () => {
      const cwd = resolveFixture('single-module');
      const output = runCommand('test', { cwd });

      expect(output).toMatchSnapshot();
    });

    it('should show get matchs when using the match command', () => {
      const cwd = resolveFixture('single-module');
      const output = runCommand('match te', { cwd });

      expect(output).toMatchSnapshot();
    });

    it('should add dependents information when using the --why option', () => {
      const cwd = resolveFixture('single-module');
      const output = runCommand('--why test', { cwd });

      expect(output).toMatchSnapshot();
    });

    it('should add dependents information when using the --why option on yarn installed package', () => {
      const cwd = resolveFixture('yarn-install');
      const output = runCommand('--why import-from', { cwd });

      expect(output).toMatchSnapshot();
    });

    it('should add dependents information when using thw -w option', () => {
      const cwd = resolveFixture('single-module');
      const output = runCommand('-w test', { cwd });

      expect(output).toMatchSnapshot();
    });
  });

  describe('qnm list', () => {
    it('should show all modules in node_modules directory', () => {
      const cwd = resolveFixture('mix-modules');
      const output = runCommand('list', { cwd });

      expect(output).toMatchSnapshot();
    });

    it('should show modules mentioned in package.json', () => {
      const cwd = resolveFixture('indirect-dependencies');
      const output = runCommand('list --deps', { cwd });

      expect(output).toMatchSnapshot();
    });

    it('should --disable-colors', () => {
      const cwd = resolveFixture('indirect-dependencies');
      const output = runCommand('list --disable-colors', {
        cwd,
        env: {
          FORCE_COLOR: '1',
        },
      });

      expect(output).toMatchSnapshot();
    });

    it('should list dependencies in a yarn installed package and show "why" information', () => {
      const cwd = resolveFixture('yarn-install');
      const output = runCommand('list --why', { cwd });

      expect(output).toMatchSnapshot();
    });
  });
});
