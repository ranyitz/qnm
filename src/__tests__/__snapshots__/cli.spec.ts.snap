// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI qnm list should --disable-colors 1`] = `
"[4manotherDependency[24m
â””â”€â”€ 1.0.0

[4mdependency1[24m
â””â”€â”€ 1.0.0

[4mdependency2[24m
â””â”€â”€ 1.0.0

[4mdevDependency1[24m
â””â”€â”€ 1.0.0

[4mdevDependency2[24m
â””â”€â”€ 1.0.0

[4mdep-of-dep[24m
â”œâ”€â”¬ [90manotherDependency[39m
â”‚ â””â”€â”€ 1.0.0
â”œâ”€â”¬ [90mdependency1[39m
â”‚ â””â”€â”€ 1.0.0
â”œâ”€â”¬ [90mdependency2[39m
â”‚ â””â”€â”€ 1.0.0
â”œâ”€â”¬ [90mdevDependency1[39m
â”‚ â””â”€â”€ 1.0.0
â””â”€â”¬ [90mdevDependency2[39m
  â””â”€â”€ 1.0.0

[4mdep-of-dep-of-dep[24m
â”œâ”€â”¬ [90manotherDependency[39m
â”‚ â””â”€â”¬ [90mdep-of-dep[39m
â”‚   â””â”€â”€ 1.0.0
â”œâ”€â”¬ [90mdependency1[39m
â”‚ â””â”€â”¬ [90mdep-of-dep[39m
â”‚   â””â”€â”€ 1.0.0
â”œâ”€â”¬ [90mdependency2[39m
â”‚ â””â”€â”¬ [90mdep-of-dep[39m
â”‚   â””â”€â”€ 1.0.0
â”œâ”€â”¬ [90mdevDependency1[39m
â”‚ â””â”€â”¬ [90mdep-of-dep[39m
â”‚   â””â”€â”€ 1.0.0
â””â”€â”¬ [90mdevDependency2[39m
  â””â”€â”¬ [90mdep-of-dep[39m
    â””â”€â”€ 1.0.0

"
`;

exports[`CLI qnm list should list dependencies in a yarn installed package and show "why" information 1`] = `
"[4mimport-cwd[24m
â””â”€â”€ [37m3.0.0[39m

[4mimport-from[24m
â””â”€â”€ [37m3.0.0[39m [33m(import-cwd)[39m

[4mresolve-from[24m
â””â”€â”€ [37m5.0.0[39m [33m(import-from)[39m

"
`;

exports[`CLI qnm list should show all modules in node_modules directory 1`] = `
"[4m@scope/test[24m
â””â”€â”€ [37m1.0.0[39m

[4manother[24m
â””â”€â”€ [37m1.0.0[39m

[4mtest[24m
â”œâ”€â”€ [37m1.0.0[39m
â””â”€â”¬ [90manother[39m
  â””â”€â”€ [37m1.0.0[39m

"
`;

exports[`CLI qnm list should show modules mentioned in package.json 1`] = `
"[4mdependency1[24m
â””â”€â”€ [37m1.0.0[39m

[4mdependency2[24m
â””â”€â”€ [37m1.0.0[39m

[4mdevDependency1[24m
â””â”€â”€ [37m1.0.0[39m

[4mdevDependency2[24m
â””â”€â”€ [37m1.0.0[39m

"
`;

exports[`CLI qnm with no arguments should add dependents information when using the --why option 1`] = `
"[4mtest[24m
â””â”€â”€ [37m1.0.0[39m [33m(devDependencies, npm install test)[39m

"
`;

exports[`CLI qnm with no arguments should add dependents information when using the --why option on yarn installed package 1`] = `
"[4mimport-from[24m
â””â”€â”€ [37m3.0.0[39m [33m(import-cwd)[39m

"
`;

exports[`CLI qnm with no arguments should add dependents information when using thw -w option 1`] = `
"[4mtest[24m
â””â”€â”€ [37m1.0.0[39m [33m(devDependencies, npm install test)[39m

"
`;

exports[`CLI qnm with no arguments should show get matchs when using the match command 1`] = `
"[4m[35mte[39mst[24m
â””â”€â”€ [37m1.0.0[39m

"
`;

exports[`CLI qnm with no arguments should show the version on a single module when called with a string 1`] = `
"[4mtest[24m
â””â”€â”€ [37m1.0.0[39m

"
`;
